### PR TITLE
Gradle, Dependencies and workflows Updated

### PR DESCRIPTION
**Pull request by this-Aditya**
Fixes: #54 

Hi @yatharthranjan, I haven't updated the `radarCommonsVersion` and `radarJerseyVersion` to the most recent release because some project structure changes are breaking the project. 
I am also not sure about updating the `confluentVersion` and `kafkaVersion`. Please let me know if they need to be updated.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -20,11 +20,12 @@ jobs:
...
             # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
-            - uses: actions/checkout@v2
+            - uses: actions/checkout@v4
 
-            - uses: actions/setup-java@v1
+            - uses: actions/setup-java@v4
               with:
-                  java-version: 11
+                  distribution: temurin
+                  java-version: 17
```
Comment (yatharthranjan) [inline .github/workflows/main.yml:28]: can you also update the dockerfile

**Comment by yatharthranjan** [comment]
> I am also not sure about updating the confluentVersion and kafkaVersion. Please let me know if they need to be updated.

Yes those can be updated. You can find the compatibility matrix for those two here - https://docs.confluent.io/platform/current/installation/versions-interoperability.html 


> I haven't updated the radarCommonsVersion and radarJerseyVersion to the most recent release because some project structure changes are breaking the project.

ok -- I think those should be updated as well (and quite important as security is handled through radar-jersey). If you are having issues, please create a new issue so those can be handled too. @mpgxvii can you take a look if you have bandwidth?

File: .github/workflows/main.yml (hunk 1 above)
Comment (this-Aditya) [reply .github/workflows/main.yml:28]: Sure!

**Review by yatharthranjan — APPROVED** [review]
LGTM, thank you.

**Comment by this-Aditya** [comment]
Hi @yatharthranjan, also created a new issue for incompatible dependencies [here](https://github.com/RADAR-base/RADAR-PushEndpoint/issues/66).

**Pull request merged**